app/src/main/java/de/rochefort/childmonitor/ListenActivity.java

```diff
@@ -16,169 +16,125 @@
 import android.os.Bundle;
-import android.support.v4.app.NotificationCompat;
-import android.support.v4.app.NotificationManagerCompat;
+import android.os.IBinder;
+import android.support.v4.content.ContextCompat;
 import android.util.Log;
 import android.widget.TextView;
-
-import java.io.IOException;
-import java.io.InputStream;
-import java.net.Socket;
-import java.net.UnknownHostException;
+import android.widget.Toast;
 
 public class ListenActivity extends Activity {
-    final String TAG = "ChildMonitor";
-    // Sets an ID for the notification
-    final static int mNotificationId = 1;
-
-    String _address;
-    int _port;
-    String _name;
-    NotificationManagerCompat _mNotifyMgr;
-
-    Thread _listenThread;
-    private final int frequency = AudioCodecDefines.FREQUENCY;
-    private final int channelConfiguration = AudioCodecDefines.CHANNEL_CONFIGURATION_OUT;
-    private final int audioEncoding = AudioCodecDefines.ENCODING;
-    private final int bufferSize = AudioTrack.getMinBufferSize(frequency, channelConfiguration, audioEncoding);
-    private final int byteBufferSize = bufferSize*2;
-
-    private void streamAudio(final Socket socket, AudioListener listener) throws IllegalArgumentException, IllegalStateException, IOException {
-        Log.i(TAG, "Setting up stream");
-
-        final AudioTrack audioTrack = new AudioTrack(AudioManager.STREAM_MUSIC,
-                frequency,
-                channelConfiguration,
-                audioEncoding,
-                bufferSize,
-                AudioTrack.MODE_STREAM);
-
-        setVolumeControlStream(AudioManager.STREAM_MUSIC);
-
-        final InputStream is = socket.getInputStream();
-        int read = 0;
-
-        audioTrack.play();
-
-        try {
-            final byte [] readBuffer = new byte[byteBufferSize];
-            final short [] decodedBuffer = new short[byteBufferSize*2];
-
-            while(socket.isConnected() && read != -1 && !Thread.currentThread().isInterrupted()) {
-                read = is.read(readBuffer);
-                int decoded = CODEC.decode(decodedBuffer, readBuffer, read, 0);
-
-                if(decoded > 0) {
-                    audioTrack.write(decodedBuffer, 0, decoded);
-                    short[] decodedBytes = new short[decoded];
-                    System.arraycopy(decodedBuffer, 0, decodedBytes, 0, decoded);
-                    listener.onAudio(decodedBytes);
-                }
-            }
-        } catch (Exception e) {
-            Log.e(TAG, "Connection failed", e);
-        } finally {
-            audioTrack.stop();
-            socket.close();
+    final String TAG = "ListenActivity";
+
+    String address;
+    int port;
+    String name;
+
+    // Don't attempt to unbind from the service unless the client has received some
+    // information about the service's state.
+    private boolean shouldUnbind;
+
+    // To invoke the bound service, first make sure that this value
+    // is not null.
```

Comment:
Correct, I've fixed it. Instead of accessing the variable, I'll use 
`context.stopService(intent)`, which incidentally also fixes that the service continues running in the background when you navigate back.

Reply:
Main reason is, that I don't know if it won't get garbage collected, if I don't keep a reference around.

Reply:
Hm, it would surprise me if a service depended on an activity to prevent it from being garbage collected.